packages/edit-site/src/components/header/document-actions/index.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	__experimentalGetBlockLabel as getBlockLabel,
+	getBlockType,
+} from '@wordpress/blocks';
+import { useSelect } from '@wordpress/data';
+
+function useSecondaryText() {
+	const selectedBlock = useSelect( ( select ) => {
+		return select( 'core/block-editor' ).getSelectedBlock();
+	} );
+
+	// TODO: Handle if parent is template part too.
```

Comment:
I wonder how to go about this here.  I was looking previously and the `getBlockParents` selector didn't seem to return the template part block as a parent for its children.  That may be a complication of of the template part block referencing a separate entity? 🤔 

One idea might be to neglect looking at the selected block in this component and instead dispatching from the template part block itself since it can easily determine `isSelected || hasInnerBlockSelected`.

Reply:
Maybe we could implement a selector in the block-editor store like `getNearestInnerBlockController` or something like that which could avoid the parent state problem

Reply:
Yeah possibly...  Although if there is an inner block controller that we aren't concerned with nested inside a template part, we may need to go beyond the nearest parent controller? 🤔   Adding something like that may be useful in other places as well, but if it is messy dispatching from the block might be very simplified.

Reply:
Hm, true. I guess that design would assume that any inner block controller could show up in the header (even if created by a 3rd party)